src/Filter/Shape/Pluck.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Membrane\Filter\Shape;
+
+use Membrane\Filter;
+use Membrane\Result\Message;
+use Membrane\Result\MessageSet;
+use Membrane\Result\Result;
+
+class Pluck implements Filter
+{
+    private readonly string $fieldSet;
+    private readonly array $fieldNames;
+
+    public function __construct(string $fieldSet, string ...$fieldNames)
+    {
+        $this->fieldSet = $fieldSet;
+        $this->fieldNames = $fieldNames;
+    }
+
+    public function filter(mixed $value): Result
+    {
+        if (!is_array($value)) {
+            $message = new Message('Pluck filter requires arrays, %s given', [gettype($value)]);
+            return Result::invalid($value, new MessageSet(null, $message));
+        }
+
+        if (array_is_list($value)) {
+            $message = new Message('Pluck filter requires arrays with key-value pairs', []);
+            return Result::invalid($value, new MessageSet(null, $message));
+        }
+
+        $from = $value[$this->fieldSet] ?? null;
```

Comment:
I think PHPStan will (Eventually) be happier if this was an array instead of null as a default value. An empty array is also more semantically correct, as that's what the code below is expecting.